### PR TITLE
fix bug on TypeError: Cannot read property 'length' of null #2072

### PR DESCRIPTION
**Pull request by rama291041610**
#2072 fix bug on TypeError: Cannot read property 'length' of null.

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -384,13 +384,14 @@ https://highlightjs.org/
 
     function subMode(lexeme, mode) {
       var i, length;
-
-      for (i = 0, length = mode.contains.length; i < length; i++) {
-        if (testRe(mode.contains[i].beginRe, lexeme)) {
-          if (mode.contains[i].endSameAsBegin) {
-            mode.contains[i].endRe = escapeRe( mode.contains[i].beginRe.exec(lexeme)[0] );
+      if(mode.contains){ //Determine if the attribute exists
```
Comment (marcoscaceres) [inline src/highlight.js]: Maybe just: 

```suggestion
      if (!mode.contains) return;
```

Comment (rama291041610) [reply src/highlight.js]: Yes, it is better.

**Comment by egor-rogov** [comment]
Hmm... I think this doesn't fix the root cause, but just hiding the problem.

Error stack trace looks like this: `highlightAuto` -> `higlight` -> `processLexeme` -> `subMode`.
Now, `highlight()` function starts with `compileLanguage()`, which calls `compileMode()`, and looking at the latter I wonder how can `mode.contains` be `null`?

```
      if (!mode.contains) {
        mode.contains = [];
      }
```
It seems to me that some language somehow didn't get compiled in time. And this is what we should fix.

@rama291041610, you wrote that the error occurs sometimes. Does it happen often? Have you notice any patterns? What language do you highlight? Can you reproduce the error with some diagnostic messages added to the highlighting code?

**Comment by yyyc514** [comment]
Every mode will have a `contains`, that's setup when the language is parsed.  Would need a reproducible example here as already requested.

```
 if (!mode.contains) {
        mode.contains = [];
      }
```

It's possible this is a dependency issue (X depends on Y, but Y hasn't been compiled), but if so it should be 100% reproducible.

**Comment by yyyc514** [comment]
Closing this as we still have the open issue.

**Pull request closed without merging**